File: js-jest/__tests__/allure-simple.test.js
describe("test", () => {
  test("Allure simple test", () => {
    allure.step("Simple step", () => {});
    allure.step("Simple parent step", () => {
      allure.step("Simple nested step", () => {});
    });
    externalStep("step parameter");
  });
});

const externalStep = param =>
  allure.step("External Step", () => {
    allure.step(`External step parameter: ${param}`, () => {});
    allure.step("Simple step inside test method", () => {});
  });

File: allure-python/allure-python-pytest/test/allure_simple_test.py
import allure


@allure.title("test_allure_simple_test displayName")
def test_allure_simple_test():
    with allure.step("Simple step"):
        pass

    with allure.step("Simple complex step"):
        with allure.step("Simple nested step"):
            pass

    simple_step_func("function parameter")


@allure.step("Simple function with step decorator")
def simple_step_func(param):
    with allure.step(f"function parameter: {param}"):
        pass

    with allure.step("Simple step inside test function"):
        pass
